src/jit/lower.cpp

```diff
@@ -3408,6 +3408,9 @@ bool Lowering::IndirsAreEquivalent(GenTreePtr candidate, GenTreePtr storeInd)
     if (pTreeA->OperGet() != pTreeB->OperGet())
         return false;
 
+    if (candidate->gtType != storeInd->gtType)
```

Comment:
Can we hoist this up above candidate->gtGetOp1() to avoid extra computation?

Also, I ran some extra validation and it looks like we need to be more specific with the type check.  There are no unsigned stind opcodes (though there are unsigned ldind opcodes).  This means that case like...

const byte myconst = 0x10;
byte foo = 0x0;
foo |= myconst;

...will generate an unneeded indirection because the types on either side of the binary op won't match (in signedness).  I think in this case that checking the sizes the is the right way to go:

if (genTypeSize(candidate->gtType) != genTypeSize(storeInd->gtType))

/cc @CarolEidt